src/app.ts

```diff
@@ -1,68 +1,130 @@
-
-import { CloudWatchLogs, S3, Lambda } from 'aws-sdk';
-
-import { getCommonConfig, getSetRetentionConfig, getConfigureLogShippingConfig } from './config';
-import { getCloudWatchLogGroups, setCloudwatchRetention, subscribeGroups, unsubscribeGroups } from "./cloudwatch";
-import { updateStructuredFieldsData } from './structuredFields';
+import { CloudWatchLogs, Lambda, S3 } from "aws-sdk";
+import {
+  getCloudWatchLogGroups,
+  setCloudwatchRetention,
+  subscribeGroups,
+  unsubscribeGroups,
+} from "./cloudwatch";
+import {
+  getCommonConfig,
+  getConfigureLogShippingConfig,
+  getSetRetentionConfig,
+} from "./config";
+import { updateStructuredFieldsData } from "./structuredFields";
 
 const { awsConfig } = getCommonConfig();
 
 const cloudwatchLogs = new CloudWatchLogs(awsConfig);
 const s3 = new S3(awsConfig);
 const lambda = new Lambda(awsConfig);
 
-function sleep(ms: number){
-    return new Promise(resolve=>{
-        setTimeout(resolve, ms)
-    })
+function sleep(ms: number) {
+  return new Promise((resolve) => {
+    setTimeout(resolve, ms);
+  });
 }
 
 export async function setRetention(): Promise<void> {
-    const { retentionInDays } = getSetRetentionConfig();
+  const { retentionInDays } = getSetRetentionConfig();
 
-    const cloudwatchLogGroups = await getCloudWatchLogGroups(cloudwatchLogs);
-    for(const logGroup of cloudwatchLogGroups) {
-        if (logGroup.retentionInDays === retentionInDays) {
-            console.log(`Log group ${logGroup.logGroupName} retention is already ${retentionInDays} days`);
-        } else {
-            await setCloudwatchRetention(cloudwatchLogs, logGroup.logGroupName!, retentionInDays);
-            // avoid hitting the SDK throttling limit
-            await sleep(200);
-            console.log(`Set ${logGroup.logGroupName} retention to ${retentionInDays} days`);
-        }
+  const cloudwatchLogGroups = await getCloudWatchLogGroups(cloudwatchLogs);
+  for (const logGroup of cloudwatchLogGroups) {
+    if (logGroup.retentionInDays === retentionInDays) {
+      console.log(
+        `Log group ${
+          logGroup.logGroupName ?? "UNDEFINED"
+        } retention is already ${retentionInDays} days`
+      );
+    } else {
+      await setCloudwatchRetention(
+        cloudwatchLogs,
+        logGroup.logGroupName!,
+        retentionInDays
+      );
+      // avoid hitting the SDK throttling limit
+      await sleep(200);
+      console.log(
+        `Set ${
+          logGroup.logGroupName ?? "UNDEFINED"
```

Comment:
`logGroup.logGroupName` is `string | undefined`, so this ensures something is rendered.